communication-service/index.js

```diff
@@ -41,7 +44,7 @@ router.get('/read', readChats);
 
 app.use('/api/chat', router).all((_, res) => {
```

Comment:
Not too sure, but I think with `cors` middleware enable above, we need not to set the headers manually anymore.

Reply:
Tested by removing it and deploy again. Not too sure why but it will cause infinite error generating in console log although I still can send message

Reply:
Have you tried just keeping the content-type header and removing the Access-Control-Allow-Origin header?

Reply:
Wow it works indeed, I will now remove the header for `Access-Control-Allow-Origin `only. Thanks for spotting this redundant code!